pkg/autodetect/main.go

```diff
@@ -147,6 +151,44 @@ func (b *Background) detectClusterRoles() {
 	}
 }
 
+func (b *Background) cleanDeployments() {
+	log.Debug("detecting orphaned deployments.")
+	deployments := &appsv1.DeploymentList{}
+	deployOpts := &client.ListOptions{}
+	jaegerOpts := &client.ListOptions{}
+	instances := &v1.JaegerList{}
+
+	// Only select fields that have the label 'sidecar.jaegertracing.io/inject'
+	if err := deployOpts.SetLabelSelector(inject.Annotation); err != nil {
+		log.WithError(err).Error("")
+	}
+
+	if err := b.cl.List(context.Background(), deployOpts, deployments); err != nil {
+		log.WithError(err).Error("Error cleaning orphaned deployment")
+	}
+
+	// get all jaeger instances
+	if err := b.cl.List(context.Background(), jaegerOpts, instances); err != nil {
+		log.WithError(err).Error("Error cleaning orphaned deployment")
+	}
+
+	// check deployments to see which one needs to be cleaned.
+	for _, dep := range deployments.Items {
+		if _, ok := dep.Annotations[inject.Annotation]; ok {
```

Comment:
IIRC, we replace the value of the annotation to include the instance name. If you store the Jaeger instances in a map indexed by the name, you can check whether the instance is in the map, instead of running the `#Select` logic.

Reply:
True! I thought we preserve the true/false in case of one jaeger instance OR the name, but you're right, that value is replaced by the instance name for all cases!

I've already done the change